xgboost_ray/main.py

```diff
@@ -257,16 +245,12 @@ def __init__(self,
 
         _set_omp_num_threads()
 
-    @property
-    def checkpoint_file(self) -> Optional[str]:
-        return _checkpoint_file(self.checkpoint_path, self.checkpoint_prefix,
-                                self.rank)
-
     @property
     def _save_checkpoint_callback(self):
         def callback(env):
-            if env.iteration % self.checkpoint_frequency == 0:
-                env.model.save_model(self.checkpoint_file)
+            if self.rank == 0 and \
+               env.iteration % self.checkpoint_frequency == 0:
+                put_queue(_Checkpoint(env.iteration, pickle.dumps(env.model)))
```

Comment:
should we use something more efficient or put it into the object store?

Reply:
Does it also work on GPUs?

Reply:
It should work on GPUs, I'll try this though.

For the object store, should we `ray.put` the model into the object store and pass the reference via the queue?
Also, it would be great if you could explain why the current approach is inefficient, so that I gather some intuition about this. Thanks!

Reply:
Yes! so from what i understand, the `model` is essentially a wrapper around some numerical datatype (array). Pickle serialization of numerical datatypes is often inefficient. Rather, the Plasma object store has much better suited for transferring numerical datatypes, so that's why `ray.put` is better! 

See: https://arrow.apache.org/blog/2017/10/15/fast-python-serialization-with-ray-and-arrow/

Reply:
Got it, thanks! I'll address this in a follow-up PR though to unblock the tune checkpointing

Reply:
Actually, I'll look into this before merging, it might be related to the error we currently see on the latest release